backend-api/tests/api-tests/clientCredentialsGrantFlow.test.ts

```diff
@@ -59,4 +172,211 @@ describe("POST /credential", () => {
       expect(response.status).toBe(401);
     });
   });
+
+  describe("Given the Bearer token in the Authorization header is invalid", () => {
+    it("Returns a 400 Bad Request response", async () => {
+      const response = await axiosInstance.post(
+        `${apiBaseUrl}/async/credential`,
+        credentialRequestBody,
+        {
+          headers: {
+            "X-Custom-Auth": "Bearer invalid.access.token",
+          },
+        },
+      );
+
+      expect(response.data).toStrictEqual({
+        error: "invalid_token",
+        error_description: "JWT payload not valid JSON",
+      });
+      expect(response.status).toBe(400);
+    });
+  });
+
+  describe("Given the request body is invalid", () => {
+    it("Returns a 400 Bad Request response", async () => {
+      const response = await axiosInstance.post(
+        `${apiBaseUrl}/async/credential`,
+        "invalidRequestBody",
+        {
+          headers: {
+            "X-Custom-Auth": "Bearer " + accessToken,
+          },
+        },
+      );
+
+      expect(response.data).toStrictEqual({
+        error: "invalid_request",
+        error_description: "Request body validation failed",
+      });
+      expect(response.status).toBe(400);
+    });
+  });
+
+  describe("Given the access token signature could not be verified", () => {
+    it("Returns 500 Server Error", async () => {
+      const accessTokenWithInvalidSignature = makeSignatureUnverifiable(
+        accessToken,
+        "6T5a8kCTyXsmw_2ATkyPgtLRzsuot-_ZIXWnuXNftZP8SHHkNxwFyMaZxEnqqtQst-99AoRrUDZnPov0oztbSA",
+      );
+
+      const response = await axiosInstance.post(
+        `${apiBaseUrl}/async/credential`,
+        credentialRequestBody,
+        {
+          headers: {
+            "X-Custom-Auth": "Bearer " + accessTokenWithInvalidSignature,
+          },
+        },
+      );
+
+      expect(response.data).toStrictEqual({
+        error: "server_error",
+        error_description: "Server Error",
+      });
+      expect(response.status).toBe(500);
+    });
+  });
+
+  describe("Given the request is valid and an active session is found", () => {
+    it("Returns 200 OK", async () => {
+      const response = await axiosInstance.post(
+        `${apiBaseUrl}/async/credential`,
+        credentialRequestBody,
+        {
+          headers: {
+            "X-Custom-Auth": "Bearer " + accessToken,
+          },
+        },
+      );
+
+      expect(response.data).toStrictEqual({
+        "https://vocab.account.gov.uk/v1/credentialStatus": "pending",
+        sub: "testSub",
+      });
+      expect(response.status).toBe(200);
+    });
+  });
+
+  describe("Given the request is valid but there is no active session", () => {
+    it("Returns 201 Created", async () => {
+      await deleteSessionIfExists(credentialRequestBody.sub);
```

Comment:
I don't think we need this step to test the scenario - but let's chat

Reply:
@jamesamooney

Reply:
Removed as discussed twice.